### DISCUSSION TITLE
MEROSS_LAN really without cloud access

### DISCUSSION THREAD

**Discussion by opu1000** [krahabb/meross_lan#466] [General]
Hello, I have installed the MEROSS_LAN addon via HACS. It works fine. I had to get the device key over the MEROSS Cloud. I do not want that any data is going into the MEROSS Cloud. Does the  MEROSS-LAN addon really with without any cloud connection?
Regards opu

**Comment by krahabb** [comment]
Hello @opu1000,
It depends:
- the HTTP protocol only works locally (it depends on your LAN but it is anyway a direct connection to your device IP)
- the MQTT protocol connects to the broker configured in the device so, if your device is paired to the Meross app/cloud, it will in turn use a connection to the Meross cloud servers.

In `meross_lan`, the connection to the cloud brokers is mediated by the 'Meross cloud profile' entry so, if you disable it, the connection via MQTT will be disabled too (this works at an higher level than simply disabling the MQTT protocol in the device config)
You can also remove it: the next time you'll configure a device you will need to provide the correct key again and so `meross_lan` could propose you to connect again...)

Without the MQTT mediation provided by the Meross cloud profile, `meross_lan` will not use cloud MQTT (but the device still does..it works by itself this way..)

**Reply by opu1000** [reply]
Thanks for the answer. I did 2 things. I did not check the box in the general Meross configuration for MQTT and I set the HTTP for the device itself. I hope this helps
![mqtt_meross2](https://github.com/krahabb/meross_lan/assets/81110967/4fa4c7de-83d6-4e56-b107-ce6c7bae5df3)
![mqtt_meross1](https://github.com/krahabb/meross_lan/assets/81110967/caa2217e-4170-47f2-947a-ced60bcae0db)

**Reply by krahabb** [reply]
This is ok: meross_lan will not use MQTT to communicate with the device at all.
But:
- keep in mind the 'Meross cloud profile' could anyway open a connection to the broker even if it's not publishing anything (it just listens incoming MQTT messages but since the device has MQTT disabled this will not create any subscription). This is because the aforementioned cloud profile, when loaded, does some management by querying the Meross api (via HTTP) for devices registrations in order to be able to detect when a new device is added to your Meross account (paired by the app)

**Comment by opu1000** [comment]
Thanks agaon for the remark. If I would delete the Meross cloud profile, does the installed MEROSS device still work? I could reinstall it, if have a 2. MEROSS device to install it in HA. Regards OPU

**Reply by krahabb** [reply]
sure, the device, once configured with the correct device key pulled from the cloud account, saves it in its own config and can live alone by connecting via HTTP

**Comment by opu1000** [comment]
thanks, this the way I will use.
